kafka/consume.go

```diff
@@ -73,29 +75,83 @@ func NewConsumer(log logrus.FieldLogger, conf Config, opts ...ConfigOpt) (*Confl
 		return nil, err
 	}
 
-	confluenceConsumer := &ConfluentConsumer{
-		c:    consumer,
-		conf: conf,
-		log:  log,
+	if conf.Timeout == 0 {
+		conf.Timeout = DefaultTimout
 	}
 
-	return confluenceConsumer, nil
+	part := int32(-1)
+	if conf.Consumer.Partition != nil {
+		part = *conf.Consumer.Partition
+	}
+
+	cc := &ConfluentConsumer{
+		c:         consumer,
+		conf:      conf,
+		log:       log,
+		partition: part,
+	}
+
+	if cc.partition < 0 {
+		// if we are not reading from a specific partition, we subscribe
+		cc.configOnce.Do(func() {
```

Comment:
Why do we need a `Once` here? It seems like we aren't using it anywhere else than in this